posthog/hogql_queries/experiments/experiment_query_runner.py

```diff
@@ -219,7 +219,7 @@ def _get_experiment_query(self) -> ast.SelectQuery:
                             constraint=ast.JoinConstraint(
                                 expr=ast.CompareOperation(
                                     left=ast.Field(chain=[metric_config.table_name, metric_config.distinct_id_field]),
-                                    right=ast.Field(chain=["exposure_data", "distinct_id"]),
+                                    right=parse_expr("toString(exposure_data.aggregation_target)"),
```

Comment:
Maybe add a comment to why this has to be cast to a string? I was not immediately clear to me why this couldn't just be `ast.Field(chain=["exposure_data", "aggregation_target"])`.